server/eclipse-project/src/main/java/au/org/democracydevelopers/corla/endpoint/EstimateSampleSizes.java

```diff
@@ -1,73 +1,258 @@
 package au.org.democracydevelopers.corla.endpoint;
 
+import au.org.democracydevelopers.corla.model.ContestType;
+import au.org.democracydevelopers.corla.model.IRVComparisonAudit;
 import spark.Request;
 import spark.Response;
 import us.freeandfair.corla.asm.ASMEvent;
+import us.freeandfair.corla.controller.ContestCounter;
 import us.freeandfair.corla.endpoint.AbstractDoSDashboardEndpoint;
 
 import org.apache.log4j.LogManager;
 import org.apache.log4j.Logger;
+import us.freeandfair.corla.model.*;
+import us.freeandfair.corla.persistence.Persistence;
+import us.freeandfair.corla.query.CountyContestResultQueries;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.apache.commons.text.StringEscapeUtils.escapeCsv;
+import static us.freeandfair.corla.controller.ComparisonAuditController.createAuditOfCorrectType;
 
 /**
- * Non-functional stub.
+ * Estimate Sample sizes.
+ * This uses ComparisonAudit::estimatedSampleSize() to get the estimated samples for each contest.
+ * It can be used either before or during an audit because it makes transient ComparisonAudits (not
+ * stored in the database).
  */
 public class EstimateSampleSizes extends AbstractDoSDashboardEndpoint {
 
-    /**
-     * Class-wide logger
-     */
-    private static final Logger LOGGER = LogManager.getLogger(EstimateSampleSizes.class);
+  /**
+   * Class-wide logger
+   */
+  private static final Logger LOGGER = LogManager.getLogger(EstimateSampleSizes.class);
 
-    /**
-     * The event to return for this endpoint.
-     */
-    private final ThreadLocal<ASMEvent> my_event = new ThreadLocal<ASMEvent>();
+  /**
+   * The event to return for this endpoint.
+   */
+  private final ThreadLocal<ASMEvent> my_event = new ThreadLocal<ASMEvent>();
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public EndpointType endpointType() {
-        return EndpointType.GET;
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public EndpointType endpointType() {
+    return EndpointType.GET;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public String endpointName() {
+    return "/estimate-sample-sizes";
+  }
+
+  /**
+   * @return STATE authorization is necessary for this endpoint.
+   */
+  public AuthorizationType requiredAuthorization() {
+    return AuthorizationType.STATE;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  protected ASMEvent endpointEvent() {
+    return my_event.get();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  protected void reset() {
+    my_event.set(null);
+  }
+
+  /**
```

Comment:
For consistency, we probably want to have Total Ballots.

In comment, "total ballots" is written all lower case.